azure-kusto-ingest/azure/kusto/ingest/_kusto_ingest_client.py

```diff
@@ -27,69 +29,124 @@ def __init__(self, kcsb):
         """
         kusto_client = KustoClient(kcsb)
         self._resource_manager = _ResourceManager(kusto_client)
+    
+    def ingest_from_dataframe(self, df, ingestion_properties):
+        blobs = []
 
-    def ingest_from_multiple_files(self, files, delete_sources_on_success, ingestion_properties):
+        file_name = 'df_{timestamp}_{pid}.csv.gz'.format(timestamp=int(time.time()), pid=os.getpid())
+        temp_file_path = os.path.join(tempfile.gettempdir(), file_name)
+
+        df.to_csv(temp_file_path, index=False, encoding='utf-8', header=False, compression='gzip')
+
+        fd = FileDescriptor(temp_file_path)
+
+        blob_name = "{db}__{table}__{guid}__{file}".format(
+            db=ingestion_properties.database,
+            table=ingestion_properties.table,
+            guid=uuid.uuid4(),
+            file=file_name,
+        )
+
+        containers = self._resource_manager.get_containers()
+        container_details = random.choice(containers)
+        storage_client = CloudStorageAccount(
+            container_details.storage_account_name, sas_token=container_details.sas
+        )
+        blob_service = storage_client.create_block_blob_service()
+
+        blob_service.create_blob_from_path(
+            container_name=container_details.object_name,
+            blob_name=blob_name,
+            file_path=temp_file_path            
+        )                
+
+        url = blob_service.make_blob_url(
+            container_details.object_name,
+            blob_name,
+            sas_token=container_details.sas,
+        )
+
+        blobs.append(BlobDescriptor(url, fd.size))
+
+        self.ingest_from_blobs(
+            blobs,
+            ingestion_properties=ingestion_properties
+        )
+
+        fd.delete_files()
+        os.unlink(temp_file_path)
+
+    def ingest_from_files(
```

Comment:
ingest_from_file , as agreed in the meeting we'll support only one blob at a time, as effectively this is what we actually do.